resource/api.go

```diff
@@ -126,18 +125,24 @@ func (a API) MarshalJSON() ([]byte, error) {
 	return json.Marshal(a.String())
 }
 
-// UnmarshalJSON parses namespace:type:subtype strings to the full API struct.
+// UnmarshalJSON parses either a string of the form namespace:type:subtype or a json object into an
+// API object.
 func (a *API) UnmarshalJSON(data []byte) error {
-	stStr := strings.Trim(string(data), "\"'")
-	if apiRegexValidator.MatchString(stStr) {
-		matches := apiRegexValidator.FindStringSubmatch(stStr)
+	var apiStr string
+	if err := json.Unmarshal(data, &apiStr); err == nil {
```

Comment:
[q] why are you unmarshaling into a string instead of using `string(data)`? Does `string` panic or something when the passed in bytes do not represent a string?

Reply:
I forgot to publish!

`string(data)` does not panic. Now that I've published, let me know if the explanation helps.

Reply:
Maybe put in a different perspective:
The combination of trimming quotes and using the regex can result in "not an api string" for two reasons:
* The input type wasn't a json string, but rather a json object
* The input type was a json string, but it wasn't formatted as `namespace:type:subtype`.

Because the old code conflates the two cases we get the confusing "cannot unmarshal into map[string]string" error when trying to parse `"api":` "not a valid api type"`.

Using `json.Unmarshal(data, &stringType)` lets me distinguish (without ambiguity) whether I'm working on a json string or a json object. So if the regex fails, we get an error message acknowledging we tried passing in a string, but the string wasn't formatted as expected.

Reply:
That makes sense! Thanks for the explanation here and in the above comment.